app/src/main/java/com/besome/sketch/editor/manage/library/firebase/ManageFirebaseActivity.java

```diff
@@ -127,15 +142,47 @@ public void onClick(View view) {
         int id = view.getId();
         if (id == R.id.btn_console) {
             goToConsole();
-        } else if (id == R.id.layout_switch) {
-            libSwitch.setChecked(!libSwitch.isChecked());
-            if ("Y".equals(firebaseLibraryBean.useYn) && !libSwitch.isChecked()) {
-                configureLibraryDialog();
+        } 
+        
+        if (id == R.id.layout_switch) {
+            //Enable Disable Firebase
+            if (libSwitch.isChecked() || !firebaseLibraryBean.data.isEmpty()) {
+
+                libSwitch.setChecked(!libSwitch.isChecked());
+                if ("Y".equals(firebaseLibraryBean.useYn) && !libSwitch.isChecked()) {
+                    configureLibraryDialog();
+                } else {
+                    firebaseLibraryBean.useYn = "Y";
+                }
+
             } else {
-                firebaseLibraryBean.useYn = "Y";
+                SketchwareUtil.toastError("Please Configure Your Firebase Project Details First," +
+                        " Either Using Settings Icon Or Importing From google-services.json");
             }
+        switch (view.getId()) {
```

Comment:
Why are you checking for the view ID again even though it's R.id.layout_switch?

Reply:
Hels , I'm 100% sure I've removed these

Reply:
Happened while merging

Reply:
Anyway just remove the switch

Reply:
ok lol